app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+                description: 'Multi select option',
+                # defaultValue: [],
+                choices: [
+                  'Land/water protection',
+                  'Land/water management',
+                  'Species management',
+                  'Education & awareness',
+                  'Law & policy',
+                  'Livelihood, economic & other incentives',
+                  'Other'
+                ],
+                otherText: 'Other'
+              },
+              {
+                type: 'dropdown',
+                name: 'implementationYear',
+                title: 'Please state the first year of commitment actions',
+                # defaultValue: [],
+                choices: %w[
+                  2022
+                  2021
+                  2020
+                  2019
+                  2018
+                  2017
+                  2016
+                  2015
+                  2014
+                  2013
+                  2012
+                ]
+              },
+              {
+                type: 'checkbox',
+                name: 'threatIds',
+                title:
+                  'What are the main threats that your commitment is seeking to address?',
+                description: 'Multi select option',
+                # defaultValue: [],
+                choices: [
+                  'Residential & commercial development',
+                  'Agriculture & aquaculture',
+                  'Energy production & mining',
+                  'Transportation & service corridors',
+                  'Biological resource use',
+                  'Human intrusions & disturbance',
+                  'Natural system modifications',
+                  'Invasive & other problematic species, genes & diseases',
+                  'Pollution',
+                  'Geological events',
+                  'Climate change & severe weather',
+                  'Other'
+                ]
+              },
+              {
+                type: 'paneldynamic',
+                name: 'linksAttributes',
+                title:
+                  'Please provide the name and URL of any other websites containing information on your commitment.',
+                description: '(Optional field)',
+                templateElements: [
+                  {
+                    type: 'text',
+                    name: 'linkName',
+                    titleLocation: 'hidden',
+                    placeHolder: 'Website name'
+                  },
+                  {
+                    type: 'text',
+                    name: 'linkUrl',
+                    titleLocation: 'hidden',
+                    placeHolder: 'https://www.example.com'
+                  }
+                ],
+                panelCount: 1,
+                confirmDelete: true,
+                confirmDeleteText: 'Are you sure you want to delete this website?',
+                panelAddText: 'Add another'
```

Comment:
not sure if it's for this object or this PR, but will need a section for records that are already saved

Reply:
yeah, I didn't know what the structure will be but I wasn't going to include the prepopulated values here because we don't have it all yet I think? But let me know if you want me to do them as wel :)